### PR TITLE
✨ Add label to run e2e on demand ✨

### PR DESCRIPTION
**Pull request by Cloud11PL**
## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [x] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #
https://linear.app/saleor/issue/MERX-766/add-label-to-run-e2e-to-be-not-run-on-every-commit
## Usage Instructions, Screenshots, Recordings

This PR introduces a label-based mechanism to trigger Playwright end-to-end (E2E) tests. Instead of running E2E tests on every commit, they will now only be executed when the pull request is labeled with run pw-e2e.

This approach aims to optimize our CI pipeline by reducing unnecessary test runs and allowing more control over when E2E tests are executed, ensuring they are triggered only when needed.

## How to use?
Apply the label run pw-e2e to the pull request when you want the E2E tests to run.
Without the label, E2E tests will be skipped.

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

## [Optional] Description
Key Changes:
- Updated the GitHub Actions workflow to conditionally run Playwright E2E tests based on the presence of the run pw-e2e label.
- Added conditional logic to skip E2E tests unless the label is present.
- Ensured that the merge of E2E test reports only occurs if the label is applied and tests have run.

### CODE REVIEW

**Review by NyanKiyoshi — APPROVED** [review]

File: .changeset/kind-weeks-kiss.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"saleor-dashboard": patch
+---
+
+Add e2e label to be not run on every commit
```
Comment (andrzejewsky) [inline .changeset/kind-weeks-kiss.md:5]: _E2e tests are no longer executed on every commit, this means performing e2e tests happens only when you add label "run pw-e2e"_

File: .github/workflows/pr-automation.yml [hunk 2]
```diff
@@ -174,7 +176,8 @@ jobs:
           PW_RETRIES: ${{ vars.PW_RETRIES }}
 
   merge-reports:
-    if: "!cancelled()"
+    if: "!cancelled() && contains(github.event.pull_request.labels.*.name, 'run pw-e2e')"
```
Comment (andrzejewsky) [inline .github/workflows/pr-automation.yml:179]: is it needed? there is "needs" that points to "run-tests", so the constraint to run this job if "run-tests" is finished is already here

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-automation.yml (hunk 2 above)
Comment (michalina-graczyk) [reply .github/workflows/pr-automation.yml]: Fixed, you're right.

File: .github/workflows/pr-automation.yml [hunk 3]
```diff
@@ -174,7 +176,8 @@ jobs:
           PW_RETRIES: ${{ vars.PW_RETRIES }}
 
   merge-reports:
-    if: "!cancelled()"
+    if: !cancelled()
```
Comment (Cloud11PL) [inline .github/workflows/pr-automation.yml:179]:
```suggestion
    if: "!cancelled()"
```

**Review by michalina-graczyk — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by NyanKiyoshi — APPROVED** [review]

**Pull request merged**